Mathlib/NumberTheory/EllipticDivisibilitySequence.lean

```diff
@@ -208,17 +245,52 @@ lemma preNormEDS_even (m : ℕ) : preNormEDS b c d (2 * (m + 3)) =
 lemma preNormEDS_neg (n : ℤ) : preNormEDS b c d (-n) = -preNormEDS b c d n := by
   rw [preNormEDS, Int.sign_neg, Int.cast_neg, neg_mul, Int.natAbs_neg, preNormEDS]
 
-/-- The canonical example of a normalised EDS `W : ℤ → R`,
-with initial values `W(0) = 0`, `W(1) = 1`, `W(2) = b`, `W(3) = c`, and `W(4) = d * b`.
+lemma preNormEDS_odd (m : ℤ) : preNormEDS b c d (2 * m + 1) =
+    preNormEDS b c d (m + 2) * preNormEDS b c d m ^ 3 * (if Even m then b else 1) -
+      preNormEDS b c d (m - 1) * preNormEDS b c d (m + 1) ^ 3 * (if Even m then 1 else b) := by
+  induction m using Int.negInduction with
+  | nat m =>
+    rcases m with _ | _ | m; simp; simp
+    simp only [Int.natCast_add, Nat.cast_one, Int.even_add_one, not_not, Int.even_coe_nat]
+    rw [Int.add_sub_cancel]
+    exact preNormEDS_odd' ..
+  | neg h m =>
+    rcases m with _ | m; simp
+    simp_rw [Int.natCast_add, Nat.cast_one, show 2 * -(m + 1 : ℤ) + 1 = -(2 * m + 1) by rfl,
+      show -(m + 1 : ℤ) + 2 = -(m - 1) by omega, show -(m + 1 : ℤ) - 1 = -(m + 2) by rfl,
+      show -(m + 1 : ℤ) + 1 = -m by omega, preNormEDS_neg, even_neg, Int.even_add_one, ite_not, h m]
+    ring1
+
+lemma preNormEDS_even (m : ℤ) : preNormEDS b c d (2 * m) =
+    preNormEDS b c d (m - 1) ^ 2 * preNormEDS b c d m * preNormEDS b c d (m + 2) -
+      preNormEDS b c d (m - 2) * preNormEDS b c d m * preNormEDS b c d (m + 1) ^ 2 := by
```

Comment:
Do you plan to use these somewhere?

Reply:
Not on top of my head. It might just be nice to have a standalone proof without using your #13057